File: e2e/php/tests/BatchTest.php
<?php

declare(strict_types=1);

namespace Kreuzberg\E2e;

use PHPUnit\Framework\TestCase;
use Kreuzcrawl\Kreuzcrawl;

/** E2e tests for category: batch. */
final class BatchTest extends TestCase
{
    /** Batch scrape of multiple URLs all succeeding */
    public function test_scrape_batch_basic(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        // skipped: field 'batch.completed_count' not available on result type
        // skipped: field 'batch.failed_count' not available on result type
        // skipped: field 'batch.total_count' not available on result type
    }

    /** Batch scrape with one URL failing returns partial results */
    public function test_scrape_batch_partial_failure(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        // skipped: field 'batch.completed_count' not available on result type
        // skipped: field 'batch.failed_count' not available on result type
        // skipped: field 'batch.total_count' not available on result type
    }

    /** Batch scrape results include specific URL */
    public function test_scrape_batch_progress(): void
    {
        $engine = Kreuzcrawl::createEngine(null);
        $result = Kreuzcrawl::scrape($engine, "");
        // skipped: field 'batch.total_count' not available on result type
        // skipped: field 'batch.results' not available on result type
    }
}

File: e2e/rust/tests/batch_test.rs
//! E2e tests for category: batch

use kreuzcrawl::scrape;
use kreuzcrawl::create_engine;

#[test]
fn test_scrape_batch_basic() {
    // Batch scrape of multiple URLs all succeeding
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let result = scrape(&engine, url).expect("should succeed");
    assert_eq!(result.batch.completed_count, "3", "equals assertion failed");
    assert_eq!(result.batch.failed_count, "0", "equals assertion failed");
    assert_eq!(result.batch.total_count, "3", "equals assertion failed");
}

#[test]
fn test_scrape_batch_partial_failure() {
    // Batch scrape with one URL failing returns partial results
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let result = scrape(&engine, url).expect("should succeed");
    assert_eq!(result.batch.completed_count, "2", "equals assertion failed");
    assert_eq!(result.batch.failed_count, "1", "equals assertion failed");
    assert_eq!(result.batch.total_count, "3", "equals assertion failed");
}

#[test]
fn test_scrape_batch_progress() {
    // Batch scrape results include specific URL
    let engine = kreuzcrawl::create_engine(None).expect("handle creation should succeed");
    let url = String::new();
    let result = scrape(&engine, url).expect("should succeed");
    assert_eq!(result.batch.total_count, "2", "equals assertion failed");
    assert!(result.batch.results.contains(r#"/target"#), "expected to contain: {}", r#"/target"#);
}

